### PR TITLE
Renamed concept type

### PR DESCRIPTION
**Pull request by pamzey**
Removed HTS prefix to all encounter type

### CODE REVIEW

**Review by Skipper-116 — CHANGES_REQUESTED** [review]
@pamzey Can you point this request to HTS branch

File: app/services/hts_service/workflow_engine.rb [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+    # Encounter types
+    INITIAL_STATE = 0 # Start terminal for encounters graph
+    END_STATE = 1 # End terminal for encounters graph
+    PREGNANCY_STATUS = 'PREGNANCY STATUS'
+    CIRCUMCISION = 'CIRCUMCISION'
+    SOCIAL_HISTORY = 'SOCIAL HISTORY'
+    SCREENING = 'SCREENING'
+    APPOINTMENT = 'APPOINTMENT'
+    HTS_Contact = 'HTS Contact'
```
Comment (andie23) [inline app/services/hts_service/workflow_engine.rb:47]: Put this in capitals like the other encounters

File: app/services/hts_service/workflow_engine.rb [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    REFERRAL = 'REFERRAL'
+    PARTNER_RECEPTION = 'Partner Reception'
+
+    # Encounters graph
+    ENCOUNTER_SM = {
```
Comment (andie23) [inline app/services/hts_service/workflow_engine.rb:52]: Encounter graph is missing HTS_CONTACT workflow

File: app/services/hts_service/workflow_engine.rb [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+      PREGNANCY_STATUS => CIRCUMCISION,
+      CIRCUMCISION => SOCIAL_HISTORY,
+      SOCIAL_HISTORY => SCREENING,
+      SCREENING => APPOINTMENT,
+      APPOINTMENT => REFERRAL,
+      REFERRAL => PARTNER_RECEPTION,
+      PARTNER_RECEPTION => END_STATE
+    }.freeze
+
+    STATE_CONDITIONS = {
```
Comment (andie23) [inline app/services/hts_service/workflow_engine.rb:63]: If possible, can you have meaningful function conditions to make encounter conditions easy to identify. 

E.g

`PREGNANCY_STATUS => %I[is_female_client]`

`CIRCUMCISION => %[is_male_client]`

`SOCIAL_HISTORY => %[has_social_history]`

e.t.c

File: app/services/hts_service/workflow_engine.rb [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+                when 'PREGNANCY STATUS'
+                     return true if @patient.gender == "F" && status == false
+
+                when "CIRCUMCISION"
+                     return true if @patient.gender == "M" && status == false
+            end
+    end
+
+    def marital_status_track?
+
```
Comment (andie23) [inline app/services/hts_service/workflow_engine.rb:145]: Please check using Social History encounter and HTS Program id

File: app/services/hts_service/workflow_engine.rb [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+           concept_id = ConceptName.find_by_name('Civil status').concept_id
+           status = Observation.where(concept_id: concept_id, person_id: @patient.id).exists?
+           return true if status == false
+    end
+
+    def workflow_status_track?
```
Comment (andie23) [inline app/services/hts_service/workflow_engine.rb:151]: Can you rename this to something more meaningful.... like "task_completed_today?"

File: app/services/hts_service/workflow_engine.rb [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+                           .where('encounter_datetime BETWEEN ? AND ?', *TimeUtils.day_bounds(@date))\
+                           .order(encounter_datetime: :desc)\
+                           .first
+      return true if encounter.blank?
+
+    end
+
+     def check_hiv_results_status?
+
+       return true if true == Observation.joins(:encounter)\
```
Comment (andie23) [inline app/services/hts_service/workflow_engine.rb:164]: is it necessary to do `return true if true ===...` when `.exists?` returns a boolean?

File: app/services/hts_service/workflow_engine.rb [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+                                         .where(concept: concept('HIV status'),
+                                                person: @patient.person,
+                                                value_coded: concept('Positive').concept_id,
+                                                encounter: { program_id: @program.program_id })\
+                                         .where('obs_datetime BETWEEN ? AND ?', *TimeUtils.day_bounds(@date))
+                                         .order(obs_datetime: :desc)\
+                                         .exists?
+    end
+
+    def htn_transform(encounter_type)
```
Comment (andie23) [inline app/services/hts_service/workflow_engine.rb:174]: Please remove this. We do not handle HTN in HTS

**Review by andie23 — CHANGES_REQUESTED** [review]
Please work on removing HTN references in the workflow engine and consider fixing some of the areas highlighted in the code.

Most importantly, your workflow is missing HTS_CONTACT in workflow graph

**Pull request closed without merging**